lua/smear_cursor/color.lua

```diff
@@ -58,72 +38,95 @@ local function interpolate_colors(hex1, hex2, t)
 	return rgb_to_hex(r, g, b)
 end
 
-M.set_hl_groups = function()
+function M.clear()
+	cache = {}
+end
+
+---@param opts? {level?: number, inverted?: boolean}
+function M.get_hl_group(opts)
+	opts = opts or {}
+	local _cursor_color ---@type string?
+
+	local hl_group = ("SmearCursorNormal%s"):format(opts.inverted and "Inverted" or "", opts.level or 0)
+
+	-- Get the cursor color from the treesitter highlight group
+	-- at the cursor.
+	if cursor_color == "treesitter" and vim.b.ts_highlight then
+		local cursor = vim.api.nvim_win_get_cursor(0)
+		local ts_hl_group ---@type string?
+		for _, capture in pairs(vim.treesitter.get_captures_at_pos(0, cursor[1] - 1, cursor[2])) do
+			ts_hl_group = "@" .. capture.capture .. "." .. capture.lang
```

Comment:
We need to loop over all captures. The last one is the one we want, so no early `break` here.